### ISSUE TITLE
Index creation should work when running GatherVcfs on block compressed VCFs

### ISSUE THREAD

**Issue by vdauwera** [broadinstitute/picard#789]
### Affected tool(s)
GatherVcfs

### Affected version(s)
Latest public release version

### Description 
Attempting to merge block-compressed VCFs produces this warning:

    WARNING	2015-10-28 15:01:48	GatherVcfs	Index creation not currently supported when gathering block compressed VCFs.

And of course no index gets created.

#### Expected behavior
We want index creation to be functional for this use case.

**Comment by ffinfo** [comment]
👍

**Comment by eitanbanks** [comment]
I don't think you actually want this behavior.  It would mean that you'd need to parse the VCF records... which completely voids the entire goal of gathering block compressed VCFs (which is just a concatenation of blocks and doesn't do any parsing).

**Comment by droazen** [comment]
Yeah, I agree with @eitanbanks that this request is not possible to satisfy without defeating the purpose of the feature -- you can't index without unpacking the compressed blocks.

**Comment by tfenne** [comment]
Just to offer a different perspective....  Picard's `GatherBamFiles` does gathering by block copying in a very similar way to GatherVcfs, but also has the ability to generate the index.  The index generation is, as you would imagine, supported by decompressing and parsing the blocks while they are being copied.  However, the decompressed version of the data is only used to generate the index.  The compressed blocks are what are written to the output BAM.  So this does still save on re-encoding the SAMRecords and probably more importantly saves on re-compressing everything.  It's slower than not generating the index, but still massively faster than a non-block-copy approach.
